jobs/delorean/jenkinsfiles/branch/suite-next/Jenkinsfile

```diff
@@ -0,0 +1,98 @@
+      suiteParams.gitUrl = userInput.installationGitUrl
+      suiteParams.suiteBranch = userInput.suiteBranch
+      verifyParameters(suiteParams)
+  }
+}
+
+// Checkout remote repository
+def checkoutGitRepo(gitUrl, gitRef, credentialsID) {
+  checkout([$class: 'GitSCM', branches: [[name: gitRef]],
+            doGenerateSubmoduleConfigurations: false,
+            extensions: [],
+            submoduleCfg: [],
+            userRemoteConfigs: [[credentialsId: credentialsID, url: gitUrl]]])
+}
+
+// Get product next branches available in the installation repo
+def getProductNextBranches(gitOrg, gitRepo, sourceBranch) {
+  nextBranches = sh(returnStdout: true, script: "git ls-remote origin | grep next | awk '{print \$2}'").replaceAll("refs/heads/", "").split()
+
+  // Do not include the source branch in the next branches list
+  productNextBranches = nextBranches.minus(sourceBranch)
+  return productNextBranches
+}
+
+def gitCreateAndCheckoutBranch(branchName, pushOnCreate = false) {
+  String remoteBranchCommit = sh(returnStdout: true, script: "git ls-remote origin refs/heads/${branchName} | cut -f 1").trim()
+  if (remoteBranchCommit) {
+    sh "git push origin --delete ${branchName}"
+  }
+
+  sh "git checkout -b ${branchName}"
+
+  if (pushOnCreate) {
+    sh "git push origin ${branchName}"
+  }
+}
+
+def suiteParams = [:]
+def ghOwner = "integr8ly"
+def ghRepo = 'installation'
+def githubSSHCredentialsID = 'jenkinsgithub'
+def masterBranch = 'master'
+
+node {
+  cleanWs()
+  stage('Ensure Suite Parameters Provided') {
+    suiteParams.gitUrl = params.installationGitUrl ?: 'git@github.com:integr8ly/installation.git'
+    suiteParams.suiteBranch = params.suiteBranch
+    suiteParams.productBranches = params.productBranches // Product branch names separated by ',', if not defined = Install all products
+    verifyParameters(suiteParams)
+    println "Suite Options: ${suiteParams}"
+  }
+
+  dir('installation') {
+    stage('Fetch Installation Repo') {
+      checkoutGitRepo(suiteParams.gitUrl, masterBranch, githubSSHCredentialsID)
+    }
+
+    sshagent([githubSSHCredentialsID]) {
+      sh 'git config --global user.name "Automatron"'
+      sh 'git config --global user.email "github.eng@feedhenry.com"'
+
+      stage('Create and Checkout Suite Branch') {
+        gitCreateAndCheckoutBranch(suiteParams.suiteBranch, true)
+      }
+
+      // Merge product-next branches into suite-next branch
+      stage('Merge Product Next Branches') {
+        def productBranches = []
+        if (suiteParams.productBranches) {
+          productBranches = suiteParams.productBranches.replaceAll(" ", "").split(',')
+        } else {
+          productBranches = getProductNextBranches(ghOwner, ghRepo, suiteParams.suiteBranch)
+        }
+
+        println "[INFO] Merging product branches: ${productBranches}"
+        for (i = 0; i < productBranches.size(); i++) {        
+          try {
+            sh "git pull --no-edit origin ${productBranches[i]}"
```

Comment:
I don't think this job should do it, but just realised we really should somehow make sure that the product next branches are up to date with master before we get this far.

Reply:
The product next branches should be getting up to date with master in the [service discovery job](https://github.com/integr8ly/ci-cd/blob/master/jobs/delorean/jenkinsfiles/discovery/next/github/Jenkinsfile#L270). However, if it's not able to rebase automatically, it'll have to be done manually. The current job doesn't fail either if that happens so I've created a [PR](https://github.com/integr8ly/ci-cd/pull/104) to make that more visible